docs/resource/k8s.md

```diff
@@ -211,3 +211,10 @@ A CronJob creates Jobs on a repeating schedule.
 <!-- endsemconv -->
 
 [DocumentStatus]: https://github.com/open-telemetry/opentelemetry-specification/tree/v1.22.0/specification/document-status.md
+
+## Well-Known Labels
+
+There are certain Kubernetes [Well-Known Labels and Annotations](https://kubernetes.io/docs/reference/labels-annotations-taints/) that SHOULD align to OpenTelemetry resource attributes. Labels are typically converted to existing OpenTelemetry attributes automatically using an OpenTelemetry Collector component like the [Kubernetes Attribute Processor](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/k8sattributesprocessor).
+
+* The Well-Known Label `app.kubernetes.io/name` SHOULD align to the OpenTelemetry resource attribute `service.name`.
```

Comment:
[This label](https://kubernetes.io/docs/reference/labels-annotations-taints/#app-kubernetes-io-name) is usually used to express the generic application name, whereas `app.kubernetes.io/instance` is the [recommended label key to indicate a specific installation](https://kubernetes.io/docs/reference/labels-annotations-taints/#app-kubernetes-io-instance) of an application in k8s. The [`service.name` attr is defined](https://github.com/open-telemetry/semantic-conventions/tree/main/docs/resource#service) as the logical name of a service. I assume this is a unique name differentiating a service installation from another; could actually do with more clarification on this.

Let's say a k8s environment has multiple installations of the application `kafka`, named `kafka-a`, `kafka-b` and `kafka-c` deployed using a `kafka` helm chart ([standard label mapping for helm](https://helm.sh/docs/chart_best_practices/labels/#standard-labels) follows the same recommendation). In this case, all pods from the 3 different statefulset are likely to have `app.kubernetes.io/name = kafka` and the value of label `app.kubernetes.io/instance` will be the specific installation/usecase of the app kafka. Depending on internal conventions, users might want `service.name` to reflect the specific instance name instead of just `kafka` in this case. All this to say, there is some ambiguity here which label would map to `service.name` and imo the convention should just suggest the possible well-known labels k8s users can consider.

Reply:
From a language perspective: how about replacing `SHOULD` with `RECOMMENDED` + add the note around ambiguity around `app.kubernetes.io/instance`?